rasa/nlu/classifiers/mitie_intent_classifier.py

```diff
@@ -1,122 +1,145 @@
-        self.clf = clf
+        """Constructs a new intent classifier using the MITIE framework."""
+        self._config = config
+        self._model_storage = model_storage
+        self._resource = resource
+        self._clf = clf
 
     @classmethod
     def required_packages(cls) -> List[Text]:
+        """Lists required dependencies (see parent class for full docstring)."""
         return ["mitie"]
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-        import mitie
+    def train(self, training_data: TrainingData, model: MitieModel) -> Resource:
+        """Trains classifier.
 
-        model_file = kwargs.get("mitie_file")
-        if not model_file:
-            raise Exception(
-                "Can not run MITIE entity extractor without a "
-                "language model. Make sure this component is "
-                "preceeded by the 'MitieNLP' component."
-            )
+        Args:
+            training_data: The NLU training data.
+            model: The loaded mitie model provided by `MitieNLP`.
+
+        Returns:
+            The resource locator for the trained classifier.
+        """
+        import mitie
 
-        trainer = mitie.text_categorizer_trainer(model_file)
-        trainer.num_threads = kwargs.get("num_threads", 1)
+        trainer = mitie.text_categorizer_trainer(str(model.model_path))
+        trainer.num_threads = self._config["num_threads"]
 
         for example in training_data.intent_examples:
             tokens = self._tokens_of_message(example)
             trainer.add_labeled_text(tokens, example.get(INTENT))
 
         if training_data.intent_examples:
             # we can not call train if there are no examples!
-            self.clf = trainer.train()
-
-    def process(self, message: Message, **kwargs: Any) -> None:
-
-        mitie_feature_extractor = kwargs.get("mitie_feature_extractor")
-        if not mitie_feature_extractor:
-            raise Exception(
-                "Failed to train 'MitieFeaturizer'. "
-                "Missing a proper MITIE feature extractor."
+            clf = trainer.train()
+            self._persist(clf)
+
+        return self._resource
+
+    def process(self, messages: List[Message], model: MitieModel) -> None:
+        """Make intent predictions using `mitie`.
+
+        Args:
+            messages: The message which the intents should be predicted for.
+            model: The loaded mitie model provided by `MitieNLP`.
+        """
+        for message in messages:
+            if self._clf:
+                token_strs = self._tokens_of_message(message)
+                intent, confidence = self._clf(token_strs, model.word_feature_extractor)
+            else:
+                # either the model didn't get trained or it wasn't
+                # provided with any data
+                intent = None
```

Comment:
(I just remember that you asked where `None` intents might happen in the other review. Looks like if we added this classifier to core's NLU pipeline then this is where they might actually be created 😅 )

Reply:
haha, very good point 👍🏻 Thanks for pointing this out!